enclave_manager/avalon_enclave_manager/kme/kme_listener.py

```diff
@@ -50,31 +52,31 @@ def __init__(self, rpc_methods):
         super().__init__(rpc_methods)
 
 
-def construct_wo_req(in_data, workload_id, encryption_key):
+def construct_wo_req(in_data, workload_id, encryption_key,
+                     session_key, session_iv):
```

Comment:
Not sure why `construct_wo_req` defined in  kme listener. AFAIK KME Listener will receive the work order request and WPE will sent the work order request to KME Listener

Reply:
The construction of a work order request requires an encryption key. When the WPE requests for UniqueVerificationKey, it is yet to signup and hence does not hold an encryption key. Only after getting the UniqueVerificationKey from the KME does the WPE signup and create the related data.

Reply:
The encryption key used to encrypt inData in wo_request from WPE is the session key generated by WPE and the session key of WPE is encrypted using public encryption key of KME and this encryptedSessionKey is also part of wo_request. WPE needs to looks the worker details of KME to get the KME public encryption key

Reply:
Plausible. I did think about this. But there could be multiple workers in the KV. What would be the basis of filtering out one worker from the workers available in that case? At  this point the WPE holds only the address of the KMEListener.

Reply:
The expectation is that WPE knows about the KME worker which can handle the workload. In fact this assumption that's true for Avalon clients also i.e Avalpn clients knows which worker can handle teh workload . You could hard code the KME worker id and WPE can retrieve the details of KME worker